lib/core/singletons/lemmy_client.dart

```diff
@@ -7,9 +10,57 @@ class LemmyClient {
 
   void changeBaseUrl(String baseUrl) {
     lemmyApiV3 = LemmyApiV3(baseUrl);
+    _populateSiteInfo(); // Do NOT await this. Let it populate in the background.
   }
 
   static final LemmyClient _instance = LemmyClient._initialize();
 
   static LemmyClient get instance => _instance;
+
+  Future<void> _populateSiteInfo() async {
+    if (!_lemmySites.containsKey(instance.lemmyApiV3.host)) {
+      // Retrieve the site so we can look up metadata about it later
+      Account? account = await fetchActiveProfileAccount();
+
+      _lemmySites[instance.lemmyApiV3.host] = await instance.lemmyApiV3.run(
+        GetSite(
+          auth: account?.jwt,
+        ),
+      );
+    }
+  }
+
+  bool supportsFeature(LemmyFeature feature) {
+    if (_lemmySites.containsKey(instance.lemmyApiV3.host)) {
+      // Parse the version
+      FullSiteView site = _lemmySites[instance.lemmyApiV3.host]!;
+      Version version;
+      try {
+        version = Version.parse(site.version);
+      } catch (e) {
+        return false;
+      }
+
+      // Check the feature and return whether it's supported in this version
+      return switch (feature) {
+        LemmyFeature.sortTypeControversial || LemmyFeature.sortTypeScaled || LemmyFeature.commentSortTypeControversial => version >= Version(0, 19, 0, preRelease: ["rc", "1"]),
+      };
+    }
+
+    return false;
+  }
+
+  static final Map<String, FullSiteView> _lemmySites = <String, FullSiteView>{};
+}
+
+enum LemmyFeature {
```

Comment:
Just throwing an idea out there to see what you think: would it be better to make this a more advanced enum (similar to [FontScale](https://github.com/thunder-app/thunder/blob/develop/lib/core/enums/font_scale.dart)) where we define the version that is supported by the given feature

e.g.,

```dart
enum LemmyFeature {
  sortTypeControversial,
  sortTypeScaled,
  commentSortTypeControversial,
}

extension LemmyFeatureExtension on LemmyFeature {
  Version get minSupportedVersion {
    switch (this) {
      case LemmyFeature.sortTypeControversial:
        return Version(0, 19, 0, preRelease: ["rc", "1"]);
      case LemmyFeature.sortTypeScaled:
        return Version(0, 19, 0, preRelease: ["rc", "1"]);
      case LemmyFeature.commentSortTypeControversial:
        return Version(0, 19, 0, preRelease: ["rc", "1"]);
    }
  }
}
```

then, in the `supportsFeature` function, we can just grab the version of the instance, and compare it to the enum value `minSupportedVersion`

```dart
bool supportsFeature(LemmyFeature feature) {
  // previous logic
 
  return version >= feature.minSupportedVersion;
}
```

Reply:
I like this idea, and I think it can be even simpler. Instead of an extension, we can have a property on the enum and set it in the constructor. I added this in ba005579a29879271021ea9aed75e313c8a906a5, let me know what you think.

Reply:
Just took a look at it and it looks good to me!